src/manifolds/EssentialManifold.jl

```diff
@@ -0,0 +1,508 @@
+a quotient space of the [`Rotations`](@ref) manifold product ``\mathrm{SO}(3)^2``.
+
+Let ``R_x(θ), R_y(θ), R_x(θ) \in ℝ^{x\times 3}`` denote the rotation around the ``z``,
+``y``, and ``x`` axis in ``ℝ^3``, respectively, and further the groups
+
+```math
+H_z = \bigl\{(R_z(θ),R_z(θ))\ \big|\ θ ∈ [-π,π) \bigr\}
+```
+
+and
+
+```math
+H_π = \bigl\{ (I,I), (R_x(π), R_x(π)), (I,R_z(π)), (R_x(π), R_y(π))  \bigr\}
+```
+
+acting elementwise on the left from ``\mathrm{SO}(3)^2`` (component wise).
+
+Then the unsigned Essential manifold ``\mathcal{M}_{\text{E}}`` can be identified
+with the quotient space
+````math
+\mathcal{M}_{\text{E}} := (\text{SO}(3)×\text{SO}(3))/(H_z × H_π),
+````
+
+and for the signed Essential manifold $\mathcal{M}_{\text{Ǝ}}$, the quotient reads
+````math
+\mathcal{M}_{\text{Ǝ}} := (\text{SO}(3)×\text{SO}(3))/(H_z).
+````
+
+An essential matrix is defined as
+````math
+E = (R'_1)^T [T'_2 - T'_1]_{×} R'_2,
+````
+where the poses of two cameras $(R_i', T_i'), i=1,2,$ are contained in the space of
+rigid body transformations $SE(3)$ and the operator $[⋅]_{×}\colon ℝ^3 \to \operatorname{SkewSym}(3)$
+denotes the matrix representation of the cross product operator. For more details see [^TronDaniilidis2017].
+
+# Constructor
+    EssentialManifold(is_signed=true)
+
+Generate the manifold of essential matrices, either the signed (`is_signed=true`) or
+unsigned (`is_signed=false`) variant.
+
+[^TronDaniilidis2017]:
+    > Tron R.; Daniilidis K.; The Space of Essential Matrices as a Riemannian Quotient
+    > Manifold.
+    > SIAM Journal on Imaging Sciences (2017),
+    > DOI: [10.1137/16M1091332](https://doi.org/10.1137/16M1091332),
+    > PDF: [https://www.cis.upenn.edu/~kostas/mypub.dir/tron17siam.pdf](https://www.cis.upenn.edu/~kostas/mypub.dir/tron17siam.pdf).
+"""
+struct EssentialManifold <: AbstractPowerManifold{ℝ,Rotations{3},NestedPowerRepresentation}
+    is_signed::Bool
+    manifold::Rotations{3}
+end
+
+EssentialManifold(is_signed::Bool=true) = EssentialManifold(is_signed, Rotations(3))
+
+@doc raw"""
+    check_manifold_point(M::EssentialManifold, p; kwargs...)
+
+Check whether the matrix is a valid point on the [`EssentialManifold`](@ref) `M`,
+i.e. a Tuple of two SO(3) matrices.
+"""
+function check_manifold_point(M::EssentialManifold, p; kwargs...)
+    if length(p) != 2
+        return DomainError(
+            length(p),
+            "The point $(p) does not lie on $M, since it does not contain exactly two elements.",
+        )
+    end
+    return check_manifold_point(
+        PowerManifold(M.manifold, NestedPowerRepresentation(), 2),
+        p;
+        kwargs...,
+    )
+end
+
+"""
+    check_tangent_vector(M::EssentialManifold, p, X; check_base_point = true, kwargs... )
+
+Check whether `X` is a tangent vector to manifold point `p` on the [`EssentialManifold`](@ref) `M`,
+i.e. `X` has to be a Tuple of two `3`-by-`3` skew-symmetric matrices. The optional parameter `check_base_point`
```

Comment:
I think it has to be a two-element array, not a tuple. Similarly in docstring for `check_manifold_point`.

Reply:
Yes, sure, since it's nested it's an array.